processing/src/main/java/org/apache/druid/query/groupby/epinephelinae/column/StringGroupByColumnSelectorStrategy.java

```diff
@@ -148,7 +153,12 @@ private void initializeGroupingKeyV2Dimension(
   @Override
   public Grouper.BufferComparator bufferComparator(int keyBufferPosition, @Nullable StringComparator stringComparator)
   {
-    if (stringComparator == null || StringComparators.LEXICOGRAPHIC.equals(stringComparator)) {
+    final boolean canCompareInts =
+        capabilities != null &&
+        capabilities.hasBitmapIndexes() &&
```

Comment:
Bitmap indexes aren't required here. I don't think it hurts too much to have in practice this check here, since in the primary case where pushdown matters (direct access to a column in a regular segment) bitmap indexes will be there. But it isn't necessary.